software/firmware/libraries/comm_protocol/examples/rover20_comms.py

```diff
@@ -0,0 +1,129 @@
+#pothos V1.0
+#author: Anthony Grana
+#date: 1/10/2019
+#OSURC Mars Rover 2019-2020
+
+import sys
+import serial
+import time
+import io
+import struct
+import timeit
+from serial import tools
+
+class rover20_comms():
+    num_slaves = 0
+    adr = []
+    data = []
+    port = ''
+
+    def __init__(self,num_slaves, data, port,TO):
+        self.num_slaves = num_slaves
+        self.data = data
+        if (not (num_slaves == len(self.data))):
+            raise ValueError('number of slaves does not match initial data')
+        for i in range(self.num_slaves):
+            self.adr.append([])
+            for j in range(len(self.data[i])):
+                self.adr[i].append(j)
+        self.port = serial.Serial(port, 115200,timeout = TO)
+
+    def list_types(self):
+        for i in range(self.num_slaves):
+            for j in range(len(self.data[i])):
+                print("Slave ID: " + str(i+1) + "    adr: " + str(self.adr[i][j]) + "   data type: " + str(self.data[i][j]))
+
+    def write_single(self, ID, adr, data):
+        self.port.write(bytes([0x00]))
+        self.port.write(bytes([ID]))
+        self.port.write(bytes([0x02]))
+        self.port.write(bytes([adr]))
+        if(isinstance(data, int) and data >= -32768 and data <= 32767 and self.data[ID-1][adr] == 'int'):
+            self.port.write((data).to_bytes(2, byteorder='big'))
+        elif(isinstance(data,float) and self.data[ID-1][adr] == 'float'):
+            self.port.write(bytearray(struct.pack("f",data)))
+        elif(isinstance(data,str) and self.data[ID-1][adr] == 'chr'):
+            self.port.write(data.encode('utf-8'))
+        elif(isinstance(data, int) and data >= -2147483648 and data <= 2147483647 and self.data[ID-1][adr] == 'long'):
+            self.port.write((data).to_bytes(4, byteorder='big'))
+        else:
+            raise TypeError("Transmit Error\nThe data you're trying to send does not mach the type stored in the register\nor, the pothos protocol does not support that data type\n or the data you're trying to send is too large or too small")
+        self.port.write(bytes([0xff]))
+        while(self.port.out_waiting):
+            time.sleep(0.0001)
+        acknowledge = self.port.read(4)
+        if (not(acknowledge == (bytes(b'\xff\xff\xff\xff')))):
```

Comment:
You don't need bytes() method.
```suggestion
        if (not(acknowledge == b'\xff\xff\xff\xff')):
```